airflow/www/fab_security/sqla/manager.py

```
@@ -580,3 +595,32 @@ def remove_permission_from_role(self, role: Role, permission: Permission) -> Non
             except Exception as e:
                 log.error(c.LOGMSG_ERR_SEC_DEL_PERMROLE.format(str(e)))
                 self.get_session.rollback()
+
+    def remove_permissions_from_role(self, role: Role, permissions: Iterable[Permission]) -> None:
+        """
+        Atomically remove multiple permission pairs from a role in a single transaction.
+
+        :param role: User role containing permissions.
+        :param permissions: Object representing multiple resource->action pairs
+        """
+        role_permissions = set(role.permissions)
+        non_assigned_permissions = []
+        for permission in permissions:
+            if permission not in role_permissions:
+                non_assigned_permissions.append(permission)
+        if non_assigned_permissions:
+            raise ValueError(
+                f"Permissions {non_assigned_permissions} are not assigned to role {role.name}, "
+                "so cannot be removed from it."
+            )
+
+        for permission in permissions:
+            role.permissions.remove(permission)
+        try:
+            self.get_session.merge(role)
+            self.get_session.commit()
+            log.info(c.LOGMSG_INF_SEC_DEL_PERMROLE.format(str(permissions), role.name))
+        except Exception as e:
+            log.error(c.LOGMSG_ERR_SEC_DEL_PERMROLE.format(str(e)))
+            self.get_session.rollback()
+            raise e
```

Actually it's not aligned with other implementations in this module (mostly borrowed from FAB, as far as I understand), which either return a bool flag or just suppress the exception. To be honest, I don't completely understand why it's the case. I'd stick to "throw early, catch late" (https://softwareengineering.stackexchange.com/questions/231057/exceptions-why-throw-early-why-catch-late), a complete suppressing may be quite confusing.